pex/resolve/pep_691/model.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+import hashlib
+
+from pex.pep_440 import Version
+from pex.pep_503 import ProjectName
+from pex.resolve.resolved_requirement import Fingerprint
+from pex.sorted_tuple import SortedTuple
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Iterable, Optional, Text
+
+    import attr  # vendor:skip
+else:
+    from pex.third_party import attr
+
+
+@attr.s(frozen=True)
+class File(object):
+    _GUARANTEED_HASH_ALGORITHM_DIGEST_SIZES = {
+        algorithm: hashlib.new(algorithm).digest_size for algorithm in hashlib.algorithms_guaranteed
+    }
+
+    @classmethod
+    def _select_algorithm(cls, algorithms):
```

Comment:
This gets tested in `test_fingerprint_service.test_mixed`, but only indirectly. I could add a dedicated unit test, but I already feel bad about the size of this thing.

Reply:
> I already feel bad about the size of this thing.

It's okay. As typical, your code is well documented and tested, and easy to follow along :)

I think a unit test would be valuable, if you're willing

Reply:
Done.